gems/aws-sdk-s3/lib/aws-sdk-s3/file_downloader.rb

```diff
@@ -23,59 +20,53 @@ def initialize(options = {})
 
       def download(destination, options = {})
         @path = destination
-        @mode = options[:mode] || 'auto'
-        @thread_count = options[:thread_count] || THREAD_COUNT
-        @chunk_size = options[:chunk_size]
-        @params = {
-          bucket: options[:bucket],
-          key: options[:key]
-        }
-        @params[:version_id] = options[:version_id] if options[:version_id]
-        @on_checksum_validated = options[:on_checksum_validated]
-        @progress_callback = options[:progress_callback]
-
+        @mode = options.delete(:mode) || 'auto'
+        @thread_count = options.delete(:thread_count) || 10
+        @chunk_size = options.delete(:chunk_size)
+        @on_checksum_validated = options.delete(:on_checksum_validated)
+        @progress_callback = options.delete(:progress_callback)
+        @params = options
         validate!
 
         Aws::Plugins::UserAgent.metric('S3_TRANSFER') do
           case @mode
```

Comment:
I'm fine with string config because that's what it is now, but we should consider symbol in the new version, or do symbol now and just call to_s here. Either is fine. Symbol is more idiomatic though.